src/plugins/vis_type_vega/public/vega_view/vega_base_view.js

```diff
@@ -297,6 +298,15 @@ export class VegaBaseView {
         this._addDestroyHandler(() => tthandler.hideTooltip());
       }
 
+      ['click', 'mouseover', 'mouseout'].forEach((eventName) => {
```

Comment:
I see you have `click` and `mouseover` covered in the external action. For `mouseout`, should there be an added action to hide the tooltip I'm assuming?

Reply:
Yes, we will need it. Wanted to first get review about this way of doing things. Can add the handler pretty quick.

Reply:
Refactored logic to use signals instead which covers the click, mouseover and mouseout events